### DISCUSSION TITLE
Add metadata to mapped_column for inclusion in MappedAsDataclass

### DISCUSSION THREAD

**Discussion by notmatthancock** [sqlalchemy/sqlalchemy#10536] [Ideas]
Dataclasses fields allow a metadata dictionary ([see here](https://docs.python.org/3/library/dataclasses.html#dataclasses.field)). It would be useful if this was available in `mapped_column` that gets inserted as normal metadata in the respective dataclass field when the MappedAsDataclass mixin is used.

**Comment by zzzeek** [comment]
whether or not there's interest in adding "metadata" (I'm concerned about the name being confusing), you can use .info on SQLAlchemy mapped attributes for this purpose.   this is accepted by mapped_column() and you'd be able to retrieve them using inspect(cls).attrs .

**Reply by CaselIT** [reply]
Maybe we could make the two synonyms? So info is also placed in the metadata field of a dataclass?

**Reply by zzzeek** [reply]
i dunno it all feels like it will lead someplace bad

**Reply by CaselIT** [reply]
Regarding the naming, do type checkers implement some special handling for the metadata keywords like they do for the other ones?

**Reply by zzzeek** [reply]
metadata is [not part of pep-673](https://peps.python.org/pep-0681/#field-specifier-parameters) since it doesn't imply anything about the behavior of the class or its attributes

**Reply by CaselIT** [reply]
interesting, does this mean that passing along metadata would be considered something non-compliant with pep 673?

**Reply by zzzeek** [reply]
i dont think it's non-compliant, just irrelevant to what pep673 provides

**Reply by CaselIT** [reply]
not sure if adding an alternative, like `dc_metadata` or `metadata_dc` is any better / clearer than just having `metadata` there.

**Reply by CaselIT** [reply]
created https://github.com/sqlalchemy/sqlalchemy/issues/10674
